src/Uno.Toolkit.UI/Markup/ItemsControlBindingExtension.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Markup;
+using Microsoft.UI.Xaml.Media;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media;
+#endif
+
+namespace Uno.Toolkit.UI
+{
+	/// <summary>
+	/// Provides a binding to the closest parent ItemsControl.
+	/// This markup can be used to access the parent ItemsControl from inside of the ItemTemplate.
+	/// </summary>
+	public class ItemsControlBindingExtension : MarkupExtension
```

Comment:
The code for this one looks like a duplicate of the `AncestorBindingExtension`, could we aggregate/inherit from it?